### PR TITLE
test: Don't collapse dmesg in CI when there's a kernel call trace

### PR DESCRIPTION
**Pull request by MggMuggins**
It would have saved me some time if the dmesg hadn't been collapsed while I was chasing [this kernel bug](https://bugs.launchpad.net/ubuntu/+source/linux/+bug/2081231).

Not thinking of any situation where we wouldn't want to know about a call trace in the dmesg during CI.

### CODE REVIEW

File: test/main.sh [hunk 1]
```diff
@@ -94,9 +94,21 @@ cleanup() {
 
   if [ "${TEST_RESULT}" != "success" ]; then
     # dmesg may contain oops, IO errors, crashes, etc
-    echo "::group::dmesg logs"
+    # If there's a kernel stack trace, don't generate a collapsible group
+
+    if journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg | grep -q "kernel: Call Trace:"; then
```
Comment (simondeziel) [inline test/main.sh:99]:
```suggestion
    if journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg --grep "kernel: Call Trace:" > /dev/null; then
```

**Review by simondeziel — APPROVED** [review]
LGTM with a tiny optional nit

File: test/main.sh (hunk 1 above)
Comment (MggMuggins) [reply test/main.sh]: Ty!

**Review by simondeziel — APPROVED** [review]

File: test/main.sh [hunk 2]
```diff
@@ -94,9 +94,22 @@ cleanup() {
...
     # dmesg may contain oops, IO errors, crashes, etc
-    echo "::group::dmesg logs"
+    # If there's a kernel stack trace, don't generate a collapsible group
+
+    expandDmesg=no
+    if journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg --grep "kernel: Call Trace:" > /dev/null; then
+      expandDmesg=yes
+    fi
+
+    if [ "${expandDmesg}" == "no" ]; then
```
Comment (simondeziel) [inline test/main.sh:104]: For historical reasons, we've used the POSIX syntax of a single `=` inside `[]`. This matters less now that we use `bash` everywhere but then it seems that double `=` inside single `[` and `]` is non-standard (https://stackoverflow.com/questions/2237080/how-to-compare-strings-in-bash/2237103#2237103)

**Comment by MggMuggins** [comment]
Wasn't having any luck with `journalctl --grep`, went back to `grep -q`

**Review by simondeziel — DISMISSED** [review]

**Comment by simondeziel** [comment]
> Wasn't having any luck with `journalctl --grep`, went back to `grep -q`; should be good to go now.

Oh, sorry about that. Works here:
```
$ journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg --grep 'apparmor="DENIED" operation="open"'
Sep 23 17:12:24 kernel: audit: type=1400 audit(1727125944.612:1494): apparmor="DENIED" operation="open" class="file" profile="snap.canonical-secscan-client.secscan-client" name="/home/sdeziel/git/pylxd/" pid=84017 comm="python3" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
Sep 23 17:12:35 kernel: audit: type=1400 audit(1727125955.505:1495): apparmor="DENIED" operation="open" class="file" profile="snap.canonical-secscan-client.secscan-client" name="/home/sdeziel/git/pylxd/" pid=84090 comm="python3" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000

$ journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg  | grep 'apparmor="DENIED" operation="open"'
Sep 23 17:12:24 kernel: audit: type=1400 audit(1727125944.612:1494): apparmor="DENIED" operation="open" class="file" profile="snap.canonical-secscan-client.secscan-client" name="/home/sdeziel/git/pylxd/" pid=84017 comm="python3" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
Sep 23 17:12:35 kernel: audit: type=1400 audit(1727125955.505:1495): apparmor="DENIED" operation="open" class="file" profile="snap.canonical-secscan-client.secscan-client" name="/home/sdeziel/git/pylxd/" pid=84090 comm="python3" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
```

Oh well, whatever works :)

**Comment by MggMuggins** [comment]
Bet it would work with just `--grep 'Call Trace:'`, I was including the `kernel: ` prefix which I guess isn't part of what's being grepped.

**Comment by MggMuggins** [comment]
Updated with that change.

**Review by simondeziel — APPROVED** [review]

**Pull request merged**